hazelcast-hibernate/hazelcast-hibernate3/src/test/java/com/hazelcast/hibernate/HibernateStatisticsTestSupport.java

```diff
@@ -99,8 +103,50 @@ protected void insertDummyEntities(int count, int childCount) {
         }
     }
 
+    protected void insertDummyNonStrictRWEntities(int count, int childCount) {
+        Session session = sf.openSession();
+        Transaction tx = session.beginTransaction();
+        try {
+            for (int i = 0; i < count; i++) {
+                DummyEntityNonStrictRW e = new DummyEntityNonStrictRW((long) i, "dummy:" + i, i * 123456d, new Date());
+                session.save(e);
+                for (int j = 0; j < childCount; j++) {
+                    DummyPropertyNonStrictRW p = new DummyPropertyNonStrictRW("key:" + j, e);
+                    session.save(p);
+                }
+            }
+            tx.commit();
+        } catch (Exception e) {
+            tx.rollback();
+            e.printStackTrace();
+        } finally {
+            session.close();
+        }
+    }
+
+    protected void insertDummyReadOnlyEntities(int count, int childCount) {
+        Session session = sf.openSession();
+        Transaction tx = session.beginTransaction();
+        try {
+            for (int i = 0; i < count; i++) {
+                DummyEntityReadOnly e = new DummyEntityReadOnly((long) i, "dummy:" + i, i * 123456d, new Date());
+                session.save(e);
+                for (int j = 0; j < childCount; j++) {
+                    DummyPropertyReadOnly p = new DummyPropertyReadOnly("key:" + j, e);
+                    session.save(p);
+                }
+            }
+            tx.commit();
+        } catch (Exception e) {
+            tx.rollback();
+            e.printStackTrace();
+        } finally {
+            session.close();
+        }
+    }
+
```

Comment:
`insertDummyReadOnlyEntities` and `insertDummyNonStrictRWEntities` are almost the same. Can you make them reuse the same code base?